File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashTable = {};
    
    for(let i=0;i<nums.length;i++){
        doesItGiveTarget = target - nums[i];
        if(hashTable[doesItGiveTarget] !== undefined)
            {
                return [hashTable[doesItGiveTarget],i]
            }
        hashTable[nums[i]]=i
    }
    return []
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_ = {}
        for i in range(len(nums)):
            next_target = target - nums[i]
            if next_target in dict_:
                return [dict_[next_target],i]
            else:
                dict_[nums[i]] = i
          
                
        